packages/pwa-kit-runtime/src/utils/ssr-server/process-lambda-response.js

```diff
@@ -6,33 +6,33 @@
  */
 
 import {CONTENT_TYPE, X_ORIGINAL_CONTENT_TYPE} from '../../ssr/server/constants'
+import {getFlattenedHeadersMap} from '@h4ad/serverless-adapter'
 
 export const processLambdaResponse = (response, event) => {
     if (!response) return response
 
     // Retrieve the correlation ID from the event headers
     const correlationId = event.headers?.['x-correlation-id']
 
-    const responseHeaders = {
-        ...response.headers
-    }
+    let joinedHeaders = getFlattenedHeadersMap(response.multiValueHeaders || {}, ',', true)
+    delete response['multiValueHeaders']
 
     // Add the correlation ID to the response headers if it exists
     if (correlationId) {
-        responseHeaders['x-correlation-id'] = correlationId
+        joinedHeaders['x-correlation-id'] = correlationId
     }
 
     // If the response contains an X_ORIGINAL_CONTENT_TYPE header,
     // then replace the current CONTENT_TYPE header with it.
-    const originalContentType = response.headers?.[X_ORIGINAL_CONTENT_TYPE]
+    const originalContentType = joinedHeaders[X_ORIGINAL_CONTENT_TYPE]
```

Comment:
@kieran-sf Did you investigate if the use of X_ORIGINAL_CONTENT_TYPE header is removable? It looks like it was added from the express app to work around some issue in serverless express https://github.com/SalesforceCommerceCloud/pwa-kit/blame/develop/packages/pwa-kit-runtime/src/utils/ssr-server/process-express-response.js

Reply:
Ah ok I think this is the explanation https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa-kit-runtime/src/ssr/server/express.js#L121-L137

Seems like we should leave it as is...